sdk/core/allocator/main.cc

```diff
@@ -224,7 +224,10 @@ namespace
 			// Release the lock before sleeping
 			g.unlock();
 			Timeout smallSleep{1};
-			thread_sleep(&smallSleep);
+			if (thread_sleep(&smallSleep) < 0)
```

Comment:
Not going to block merge but would be nice to have a union type for this. Did you audit all calls to `thread_sleep`, even ones that didn't trigger the `nodiscard` warning?